app/models/miq_server/worker_management/monitor/validation.rb

```diff
@@ -4,22 +4,50 @@ module MiqServer::WorkerManagement::Monitor::Validation
   def validate_worker(w)
     return true if MiqEnvironment::Command.is_podified?
 
-    time_threshold   = get_time_threshold(w)
-    memory_threshold = get_memory_threshold(w)
+    if !worker_heartbeat_valid?(w)
+      stop_worker(w, MiqServer::NOT_RESPONDING)
+      return false
+    end
 
-    w.validate_active_messages
+    if !worker_memory_valid?(w)
+      stop_worker(w)
+      return false
+    end
 
-    validate_heartbeat(w)
+    return true unless worker_get_monitor_status(w.pid).nil?
+
+    true
+  end
+
+  def validate_active_messages(processed_worker_ids = [])
+    actives = MiqQueue.where(:state => 'dequeue').includes(:handler)
+    actives.each do |msg|
+      next if processed_worker_ids.include?(msg.handler_id)
+
+      # Exclude messages on starting/started servers
+      handler = msg.handler
+      handler_server = handler.respond_to?(:miq_server) ? handler.miq_server : handler
+      if handler_server.kind_of?(MiqServer) && handler_server != self
+        next if [MiqServer::STATUS_STARTED, MiqServer::STATUS_STARTING].include?(handler_server.status)
+      end
+
+      msg.check_for_timeout(_log.prefix)
+    end
+  end
```

Comment:
Any reason this was moved from the original location?  I don't think it changed?

Reply:
It wasn't moved that's just how the diff is showing it.

Reply:
weird...looks like a move in https://github.com/ManageIQ/manageiq/pull/19609/commits/86d4d64141eb0a6691ae22fd4d6e7a532c077788

Reply:
Yeah, I think it's because I moved so much of the original content of `validate_worker` to new methods under `validate_active_messages`. I could put the new methods above the old one and see if that helps.

Reply:
Rebased to minimize the diff. Don't think this is included anymore.